packages/create-astro/src/index.ts

```diff
@@ -208,10 +229,55 @@ export async function main() {
 				}
 			})
 		);
-	}
 
-	templateSpinner.text = green('Template copied!');
-	templateSpinner.succeed();
+		templateSpinner.text = green('Template copied!');
+		templateSpinner.succeed();
+
+		const printIgnoreUpdatingPkgNameMsg = () => info('Opps', 'You have to update the NPM package name manually.');
+		const { stdout: npmVersion } = await execa('npm', ['-v']);
+		const npmVersionMajorPart = Number(npmVersion.split('.')[0]);
+		if (npmVersionMajorPart <= 6) {
+			// Ignore updating package name if npm version is v6 and older
+			await printIgnoreUpdatingPkgNameMsg();
+		} else {
+			// Validate and update the NPM package name
+			let pkgName: string = projectDir;
+			if (args.package) {
+				pkgName = args.package;
+			}
+
+			const validatePackageNameResult = await validatePackageName(pkgName);
+			if (!validatePackageNameResult.isValid) {
+				const pkgNameResponse = await prompts(
+					{
+						type: 'text',
+						name: 'package',
+						message: 'What is the new NPM package name?',
+						initial: generateProjectName(),
+						async validate(value) {
+							const { isValid, notices } = await validatePackageName(value);
+							if (!isValid) {
+								const errorMsg = `"${bold(value)}" is an invalid NPM package name`;
+								const errors = notices.join(', ');
+								return `${errorMsg}: ${errors}`;
+							}
+							return true;
+						},
+					},
+					{ onCancel: printIgnoreUpdatingPkgNameMsg }
+				);
+				pkgName = pkgNameResponse.package as string;
+			}
+
+			if (typeof pkgName === 'string' && !!pkgName) {
+				// Change project name in package.json
+				await execaCommand(`${pkgManager} -v`, { cwd });
+				await execaCommand(`npm -v`, { cwd });
+				await execaCommand(`npm pkg set name=${pkgName}`, { cwd });
```

Comment:
Do we really need to run all three of these? I'd expect to use `pkgManager` and fail silently if `pkg set name` is not supported.

Reply:
@natemoo-re Let me check this!